File: TestandoRecyclerView/app/src/main/java/com/myapp/testandorecyclerview/MainActivity.java
package com.myapp.testandorecyclerview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RecyclerView.Adapter adapter;
    private RecyclerView.LayoutManager layoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        this.recyclerView = findViewById(R.id.recyclerview);
//        this.recyclerView.setHasFixedSize(true);

        this.recyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        this.adapter = new MyAdapter(this, getData());

        this.recyclerView.setAdapter(adapter);

    }


    private List<String> getData() {
        List<String> ls = new ArrayList<>();
        ls.add("1");
        ls.add("2");
        ls.add("3");
        ls.add("4");



        return ls;
    }

}

File: TestadoRetrofit2/app/src/main/java/com/example/fernando/testadoretrofit2/MainActivity.kt
package com.example.fernando.testadoretrofit2

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

class Usuario(var login: String, var followers: Int)
interface GithubService {
    @GET("codificador")
    fun dados():Call<Usuario>
}


class RetrofitInitializer {
    fun init():Retrofit {
        return Retrofit.Builder()
            .baseUrl("https://api.github.com/users/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
}

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        RetrofitInitializer().init().create(GithubService::class.java).dados().run {
            enqueue(object : Callback<Usuario?> {
                override fun onFailure(call: Call<Usuario?>, t: Throwable) {}

                override fun onResponse(call: Call<Usuario?>, response: Response<Usuario?>) {
                    Log.d("ok", "" + (response.body() as Usuario).followers)
                }
            })
        }
    }
}
